test/unittests/execute_floating_point_test.cpp

```diff
@@ -66,4 +66,129 @@ TEST(execute_floating_point, f64_add)
-}
\ No newline at end of file
+}
+
+template <typename T>
+class execute_floating_point_types : public testing::Test
+{
+};
+
+struct FloatingPointName
+{
+    template <typename T>
+    static std::string GetName(int);
+};
+template <>
+std::string FloatingPointName::GetName<float>(int)
+{
+    return "f32";
+}
+template <>
+std::string FloatingPointName::GetName<double>(int)
+{
+    return "f64";
+}
+
+using FloatingPointTypes = testing::Types<float, double>;
+TYPED_TEST_SUITE(execute_floating_point_types, FloatingPointTypes, FloatingPointName);
+
+TYPED_TEST(execute_floating_point_types, compare)
+{
+    using Limits = typename FP<TypeParam>::Limits;
+
+    /* wat2wasm
+    (func (param f32 f32) (result i32) (f32.eq (local.get 0) (local.get 1)))
+    (func (param f32 f32) (result i32) (f32.ne (local.get 0) (local.get 1)))
+    (func (param f32 f32) (result i32) (f32.lt (local.get 0) (local.get 1)))
+    (func (param f32 f32) (result i32) (f32.gt (local.get 0) (local.get 1)))
+    (func (param f32 f32) (result i32) (f32.le (local.get 0) (local.get 1)))
+    (func (param f32 f32) (result i32) (f32.ge (local.get 0) (local.get 1)))
+    (func (param f64 f64) (result i32) (f64.eq (local.get 0) (local.get 1)))
+    (func (param f64 f64) (result i32) (f64.ne (local.get 0) (local.get 1)))
+    (func (param f64 f64) (result i32) (f64.lt (local.get 0) (local.get 1)))
+    (func (param f64 f64) (result i32) (f64.gt (local.get 0) (local.get 1)))
+    (func (param f64 f64) (result i32) (f64.le (local.get 0) (local.get 1)))
+    (func (param f64 f64) (result i32) (f64.ge (local.get 0) (local.get 1)))
+    */
+    const auto wasm = from_hex(
+        "0061736d01000000010d0260027d7d017f60027c7c017f030d0c0000000000000101010101010a610c07002000"
+        "20015b0b0700200020015c0b0700200020015d0b0700200020015e0b0700200020015f0b070020002001600b07"
+        "0020002001610b070020002001620b070020002001630b070020002001640b070020002001650b070020002001"
+        "660b");
+    auto inst = instantiate(parse(wasm));
+
+    constexpr FuncIdx func_offset = std::is_same_v<TypeParam, float> ? 0 : 6;
+    constexpr auto eq = func_offset + 0;
+    constexpr auto ne = func_offset + 1;
+    constexpr auto lt = func_offset + 2;
+    constexpr auto gt = func_offset + 3;
+    constexpr auto le = func_offset + 4;
+    constexpr auto ge = func_offset + 5;
+
+    const TypeParam ordered_values[] = {
+        -Limits::infinity(),
+        -Limits::max(),
+        std::nextafter(-Limits::max(), TypeParam{0}),
+        std::nextafter(-TypeParam{1.0}, -Limits::infinity()),
+        -TypeParam{1.0},
+        std::nextafter(-TypeParam{1.0}, TypeParam{0}),
+        std::nextafter(-Limits::min(), -Limits::infinity()),
+        -Limits::min(),
+        std::nextafter(-Limits::min(), TypeParam{0}),
+        std::nextafter(-Limits::denorm_min(), -Limits::infinity()),
+        -Limits::denorm_min(),
+        TypeParam{0},  // FIXME.
```

Comment:
What is the fixme here?

Reply:
It was to add "negative zero" comparison.